pkg/sidecar_old/server/handlers_test.go

```diff
@@ -283,7 +257,8 @@ func TestServer_readMessageRoutine(t *testing.T) {
 				if received == tt.wantErr {
 					t.Errorf("Server_readMessageRoutine received = %v, wantErr = %v", received, tt.wantErr)
 				}
-
+			default:
+				return
```

Comment:
This is bad, your test is not testing anything, you are just returning. The context gets all fucked up, the idea of the test is that the context has a timeout.

Reply:
Actually this test isn't even mine, I changed it to be like that because the linter was complaining about the select with only one case